Library/Homebrew/utils/curl.rb

```diff
@@ -328,30 +338,30 @@ def curl_http_content_headers_and_checksum(
         user_agent:        user_agent
       )
 
-      status_code = :unknown
-      while status_code == :unknown || status_code.to_s.start_with?("3")
-        headers, _, output = output.partition("\r\n\r\n")
-        status_code = headers[%r{HTTP/.* (\d+)}, 1]
-        location = headers[/^Location:\s*(.*)$/i, 1]
-        final_url = location.chomp if location
-      end
-
       if status.success?
+        parsed_output = parse_curl_output(output)
+        heads = parsed_output[:heads]
+
+        status_code = heads.last[:status_code] if heads.last.is_a?(Hash)
```

Comment:
Don't really like type-sniffing here. Can this be a separate part of `parsed_output` instead?

Reply:
Just to make sure I understand correctly, are you suggesting this as another field on the return hash from `#parse_curl_output`? If so, it could potentially look something like this (where `:status_code` is the status code of the last response):

```ruby
data = { heads: heads, body: output }
data[:status_code] = heads.last[:status_code] if heads.last.present?
data
```

If it's not clear, `:status_code` is optional here because curl output only contains the response body by default (so `heads` will be empty) and we have to use additional options (`--include`) to include the HTTP responses at the start of the output.